### PR TITLE
Adding support for managedclusterset labelling

### PR DESCRIPTION
**Pull request by gurnben**
## Summary of Changes

This PR adds support for ACM 2.3's new ManagedClusterSets - required for creating clusterpools on ACM 2.3!

### CODE REVIEW

File: clusterpools/apply.sh [hunk 1]
```diff
@@ -740,6 +791,10 @@ fi
 if [[ "$CLUSTERPOOL_SKIP_MACHINEPOOL" == "true" ]]; then
     printf "\n  skipMachinePools: True" >> ./${CLUSTERPOOL_NAME}/${CLUSTERPOOL_NAME}.clusterpool.yaml
 fi
+# add a ManagedClusterSet if specified
+if [[ "$MANAGEDCLUSTERSET_NAME" ]]; then
```
Comment (dhaiducek) [inline clusterpools/apply.sh:795]: I didn't know this would work without flags...good to know...

File: clusterpools/apply.sh [hunk 2]
```diff
@@ -740,6 +791,10 @@ fi
...
+    MANAGEDCLUSTERSET_NAME="'cluster.open-cluster-management.io/clusterset: $MANAGEDCLUSTERSET_NAME'" yq e '.metadata.labels = [env(MANAGEDCLUSTERSET_NAME)]' -i ./${CLUSTERPOOL_NAME}/${CLUSTERPOOL_NAME}.clusterpool.yaml
```
Comment (dhaiducek) [inline clusterpools/apply.sh]: I get an error here:
```
error: unable to decode "./test-pool/test-pool.clusterpool.yaml": resource.metadataOnlyObject.ObjectMeta: v1.ObjectMeta.Labels: ReadMapCB: expect { or n, but found [, error found in #10 byte of ...|"labels":["cluster.o|..., bigger context ...|io/v1","kind":"ClusterPool","metadata":{"labels":["cluster.open-cluster-management.io/clusterset: ac|...
```
Maybe some missing quotes around the key/values?

**Review by dhaiducek — CHANGES_REQUESTED** [review]
I think as expected, I get an admission webhook error if I don't use a ManagedClusterSet:
```
Error from server (Forbidden): error when creating "./test-pool/test-pool.clusterpool.yaml": admission webhook "ocm.validating.webhook.admission.open-cluster-management.io" denied the request: user "dhaiducek" cannot add/remove the resource to/from ManagedClusterSet ""
```

But I get an unexpected error when the script applies the label (see comment).

File: clusterpools/apply.sh (hunk 2 above)
Comment (dhaiducek) [reply clusterpools/apply.sh]: Ah, no--it's expecting key/values. Not an array?

Comment (gurnben) [reply clusterpools/apply.sh]: Hmmmm - I tested this locally successfully - what yq version are you running @dhaiducek ?

Comment (gurnben) [reply clusterpools/apply.sh]: We've located and patched the issue.  Turns out yq 4.4 and 4.11 handle key-value pair creation differently (is my guess, I've upgraded and patched and validated on yq 4.12).

**Review by dhaiducek — APPROVED** [review]

**Pull request merged**